src/providers/mod.rs

```diff
@@ -0,0 +1,4 @@
+pub mod http;
+
+/// A provider returns an initial configuration string, if successful.
+pub type Result = std::result::Result<String, &'static str>;
```

Comment:
I think that `ConfigBuilder` is probably the better interface than raw strings here.

Reply:
Switched to `ConfigBuilder` in 9cc1b33cb3bb3b2b592a7ff9fb51c5df6706536c.